### PR TITLE
[@types/uuid]: Update for uuid v10

### PR DESCRIPTION
**Pull request by joshkel**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:

- [X] Provide a URL to documentation or source code which provides context for the suggested changes: [README.md diff](https://github.com/uuidjs/uuid/compare/v9.0.1...v10.0.0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5)
- [X] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

### CODE REVIEW

**Comment by joshkel** [comment]
Regarding types for the new `v6ToV1` and `v1ToV6` functions: Local testing suggest that they always return `Uint8Array` if given an array-like or buffer-like object (unlike the normal `v1`, etc., which preserve the type of the `OutputBuffer`). Therefore, a return type of `T extends OutputBuffer` is incorrect, but specifying a return type of `Uint8Array` feels like an implementation dependency (because the docs don't spell this out). Specifying a type of `((uuid: Uint8Array) => Uint8Array) & ((uuid: OutputBuffer) => OutputBuffer)` seems like a compromise.

**Comment by ctavan** [comment]
@broofa

**Review by broofa — APPROVED** [review]
Looks good to me.  Thanks for doing this @joshkel .

File: types/uuid/index.d.mts [hunk 1]
```diff
@@ -1,12 +1,19 @@
...
 export import NIL = uuid.NIL;
+export import MAX = uuid.MAX;
 export import version = uuid.version;
 export import validate = uuid.validate;
 export import stringify = uuid.stringify;
 export import parse = uuid.parse;
 export import V1Options = uuid.V1Options;
 export import V4Options = uuid.V4Options;
+export import V6Options = uuid.V7Options;
+export import V7Options = uuid.V6Options;
```
Comment (haines) [inline types/uuid/index.d.mts:19]:
```suggestion
export import V6Options = uuid.V6Options;
export import V7Options = uuid.V7ptions;
```

**Review by broofa — APPROVED** [review]

**Review by haines — APPROVED** [review]
I'll obey the bot although I'm not in any way related to the project 😅

**Comment by jaydenseric** [comment]
Can we please merge and publish? Keen to be able to migrate to `uuid` v10 in work projects.

**Comment by broofa** [comment]
@RyanCavanaugh Would it be possible to have @typescript-bot provide information on how long PRs typically sit in the [Needs Maintainer Review queue](https://github.com/DefinitelyTyped/DefinitelyTyped/projects/5#column-7552922) for.  Doing so would help set expectations.

**Comment by broofa** [comment]
@BurningEnlightenment https://github.com/uuidjs/uuid/pull/763

As for why it's taken this long... well... that's another topic, and not something we should be spamming this PR with.

**Comment by BurningEnlightenment** [comment]
@broofa I appreciate the effort to migrate to typescript. However, I proposed something different as a stopgap measure for v10: Import the `.d.ts` files from this repository into your repository and publish them with the `uuid` npm package. If you want, I can draft a PR for you.

**Comment by broofa** [comment]
@BurningEnlightenment Hmm... interesting idea.  You're welcome to put up a PR for that in `uuid` and I'll consider it for a future release (if my TS port work stalls out, for example), but in the near-term I'd like to see this PR merged and released ASAP.

**Review by gabritto — APPROVED** [review]

**Comment by joshkel** [comment]
Ready to merge

**Pull request merged**